src/coreclr/jit/hwintrinsiccodegenarm64.cpp

```diff
@@ -743,8 +744,9 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
                 {
                     // fmov reg1, reg2
                     assert(GetEmitter()->IsMovInstruction(ins));
-                    GetEmitter()->emitIns_Mov(ins, emitTypeSize(intrin.baseType), targetReg, op1Reg,
-                                              /* canSkip */ false, INS_OPTS_NONE);
+                    bool canSkip = ((targetReg == op1Reg) && (intrin.baseType == intrin.op1->gtType));
```

Comment:
This one should always be skippable. `CreateScalarUnsafe` only guarantees the value of the lowest element while `fmov` moves a scalar value and zeros the upper bits. If we're already in the right register, there is nothing needed.

Reply:
> This one should always be skippable. `CreateScalarUnsafe` only guarantees the value of the lowest element while `fmov` moves a scalar value and zeros the upper bits. If we're already in the right register, there is nothing needed.

Here, if it is skippable, then is it ok to completely skip calling emitIns_Mov() ?

Reply:
The ideal is we always call `emitIns_Mov` and we pass in the appropriate `canSkip` value. This avoids issues with paths diverging with different capabilities/logic/etc.

In this case, I expect we can just pass in `/* canSkip */ true`

Reply:
Can we add an assert for `assert(((targetReg == op1Reg) && (intrin.baseType == intrin.op1->gtType)))`?

Reply:
I'll add `assert((intrin.baseType == intrin.op1->gtType))` (as the other might not be true)

Reply:
The assert wouldn't be valid as `targetReg` may not be the same as `op1Reg`.

`canSkip` doesn't mean "should skip", it means that the `move` doesn't have side effects that need to be considered and therefore can be elided if it matches one of the conditions allowed in `emitIns_Mov` (one of which is `targetReg == op1Reg`).

We could probably assert `intrin.baseType == intrin.op1->gtType` since this takes a scalar and produces a simd node of the same type.